velox/expression/tests/SparkExpressionFuzzerTest.cpp

```diff
@@ -62,7 +62,11 @@ int main(int argc, char** argv) {
       "chr",
       "replace",
       "might_contain",
-      "unix_timestamp"};
+      "unix_timestamp",
+      "substring_index"}; // The spark substring_index sig is
+                          // substring_index(varchar, varchar, int), but the
+                          // test contain substring_index(varchar, varchar, int)
```

Comment:
Missing typing for two different signatures? And why there is mismatched function signature, because only spark sql functions are registered in this fuzzer test.
> facebook::velox::functions::sparksql::registerFunctions("");

Reply:
It seems that this problem has been fixed, and the latest commit does not have this problem. And I removed this code.

Reply:
@rui-mo I run the Fuzzer test locally. And get the following exception. It seems the substring_index sig is (varchar, varchar, varchar) and is not same with spark's function sig. Do you know how to solve this issue?
```
I0801 17:11:00.395459 447442 ExpressionVerifier.cpp:81] Executing expression 0 : substring_index(NcY|@I|AcEAT_,|Y*_)|U/<4D<?F9B_]+L?.rp^q-xu\EvrmVt}#)MHfbBXD]\4d|,rQvV;L'l"Af5VJV\;e;s0iv7YwU,1D2/YWXk/*'v-D9Z}ZhagX]0xU[cq&:%c>?ii0ww\|K,abs("c0"))
I0801 17:11:00.395471 447442 ExpressionVerifier.cpp:30] 1 vectors as input:
I0801 17:11:00.395474 447442 ExpressionVerifier.cpp:32]         [DICTIONARY INTEGER: 100 elements, 8 nulls], [DICTIONARY INTEGER: 100 elements, 12 nulls], [DICTIONARY INTEGER: 100 elements, 9 nulls], [DICTIONARY INTEGER: 100 elements, 7 nulls], [FLAT INTEGER: 100 elements, 4 nulls]
E0801 17:11:00.396211 447442 Exceptions.h:69] Line: ../../velox/expression/tests/ExpressionVerifier.cpp:61, Function:operator(), Expression: left->equalValueAt(right.get(), row, row) Different results at idx '1': '[1->80] varchar*_)|U/<4X=PVinteger'l"Af5VJ' vs. 'NcY|@I|AcEAT_,|Y*_)|U/<4D<?F9B_]+L?.rp^q-xu\EvrmVt}#)MHfbBXD]\4d|,rQvV;L'l"Af5VJV\;e;s0iv7YwU', Source: RUNTIME, ErrorCode: INVALID_STATE
I0801 17:11:00.396708 447442 ExpressionVerifier.cpp:327] Persisted input: --fuzzer_repro_path /tmp/spark_fuzzer_repro/velox_expressionVerifier_4kL55G --input_path /tmp/spark_fuzzer_repro/velox_expressionVerifier_4kL55G/input_vector --result_path /tmp/spark_fuzzer_repro/velox_expressionVerifier_4kL55G/result_vector --sql_path /tmp/spark_fuzzer_repro/velox_expressionVerifier_4kL55G/sql
terminate called after throwing an instance of 'facebook::velox::VeloxRuntimeError'
  what():  Exception: VeloxRuntimeError
Error Source: RUNTIME
Error Code: INVALID_STATE
Reason: Different results at idx '1': '[1->80]
*** Aborted at 1690909860 (Unix time, try 'date -d @1690909860') ***
*** Signal 6 (SIGABRT) (0x3e90006d3d2) received by PID 447442 (pthread TID 0x7f64a5764c00) (linux TID 447442) (maybe from PID 447442, UID 1001) (code: -6), stack trace: ***
_build/debug/velox/expression/tests/spark_expression_fuzzer_test(_ZN5folly10symbolizer17getStackTraceSafeEPmm+0x80)[0x56194dfc9fef]
_build/debug/velox/expression/tests/spark_expression_fuzzer_test(_ZN5folly10symbolizer21SafeStackTracePrinter15printStackTraceEb+0x6a)[0x56194e0eb80c]
_build/debug/velox/expression/tests/spark_expression_fuzzer_test(+0x415dc43)[0x56194e0e9c43]
_build/debug/velox/expression/tests/spark_expression_fuzzer_test(+0x415dd24)[0x56194e0e9d24]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x1441f)[0x7f64a5ce641f]
/lib/x86_64-linux-gnu/libc.so.6(gsignal+0xcb)[0x7f64a57d500b]
/lib/x86_64-linux-gnu/libc.so.6(abort+0x12a)[0x7f64a57b4858]
/lib/x86_64-linux-gnu/libstdc++.so.6(+0x9e910)[0x7f64a5b8e910]
/lib/x86_64-linux-gnu/libstdc++.so.6(+0xaa38b)[0x7f64a5b9a38b]
/lib/x86_64-linux-gnu/libstdc++.so.6(_ZSt9terminatev+0x16)[0x7f64a5b9a3f6]
/lib/x86_64-linux-gnu/libstdc++.so.6(__cxa_rethrow+0x4c)[0x7f64a5b9a6fc]
_build/debug/velox/expression/tests/spark_expression_fuzzer_test(__cxa_rethrow+0x106)[0x56194dfc5cf5]
_build/debug/velox/expression/tests/spark_expression_fuzzer_test(_ZN8facebook5velox4test16ExpressionFuzzer2goEv+0x85c)[0x56194c013a4e]
_build/debug/velox/expression/tests/spark_expression_fuzzer_test(_ZN8facebook5velox4test16expressionFuzzerESt13unordered_mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESt6vectorIPKNS0_4exec17FunctionSignatureESaISD_EESt4hashIS8_ESt8equal_toIS8_ESaISt4pairIKS8_SF_EEEm+0xab)[0x56194c013c30]
_build/debug/velox/expression/tests/spark_expression_fuzzer_test(_ZN12FuzzerRunner3runERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEmRKSt13unordered_setIS5_St4hashIS5_ESt8equal_toIS5_ESaIS5_EES7_+0x99)[0x56194bfd13f5]
_build/debug/velox/expression/tests/spark_expression_fuzzer_test(main+0x317)[0x56194bfce4d6]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf2)[0x7f64a57b6082]
_build/debug/velox/expression/tests/spark_expression_fuzzer_test(_start+0x2d)[0x56194bfcbe3d]
(safe mode, symbolizer not available)
Aborted (core dumped)
```

Reply:
I checked the implementation of spark fuzzer test, and found it should only test with the registered signatures. Maybe we can check this offline.

Reply:
@rui-mo Thanks for your help. The `fuzzer ` tests passed. Can you help to review again?